### PR TITLE
Feature/88055 export document definition + endpoint

### PR DESCRIPTION
**Pull request by marijnritense**
- New management endpoint to export a case definition
  - Creates a zip with a document definition in it, in a structure that mimics a project resource folder

### CODE REVIEW

File: case/src/main/kotlin/com/ritense/case/service/CaseDefinitionExportService.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
+
+open class CaseDefinitionExportService(
+    private val documentDefinitionExportService: JsonSchemaDocumentDefinitionExportService
+) {
+    open fun createExport(caseDefinitionId: JsonSchemaDocumentDefinitionId): ByteArrayOutputStream {
+        val exportList: Set<ExportFile> = documentDefinitionExportService.export(caseDefinitionId)
+        val baos = ByteArrayOutputStream()
+        ZipOutputStream(baos).use { zos ->
```
Comment (ivo-ritense) [inline case/src/main/kotlin/com/ritense/case/service/CaseDefinitionExportService.kt:32]: Don't we need to close the stream?

File: case/src/main/kotlin/com/ritense/case/web/rest/CaseDefinitionResource.kt [hunk 2]
```diff
@@ -148,4 +154,22 @@ class CaseDefinitionResource(
...
+        produces = [MediaType.APPLICATION_OCTET_STREAM_VALUE]
+    )
+    @RunWithoutAuthorization
+    fun getExport(
+        @PathVariable caseDefinitionName: String,
+        @PathVariable caseDefinitionVersion: Long,
+    ): ResponseEntity<ByteArray> {
+        val baos = exportService
+            .createExport(JsonSchemaDocumentDefinitionId.existingId(caseDefinitionName, caseDefinitionVersion))
+        val fileName = caseDefinitionName + "_" + caseDefinitionVersion + "_" +
```
Comment (ivo-ritense) [inline case/src/main/kotlin/com/ritense/case/web/rest/CaseDefinitionResource.kt:168]: Nitpicking: you complete the filename by adding the extension in the header. Might be a bit counter intuitive.

File: case/src/test/kotlin/com/ritense/case/web/rest/CaseDefinitionResourceIntTest.kt [hunk 3]
```diff
@@ -1168,6 +1168,21 @@ class CaseDefinitionResourceIntTest : BaseIntegrationTest() {
...
+    fun `should export case definitions as an admin`() {
+        val caseDefinitionName = "house"
+        val caseDefinitionVersion = 1
+        mockMvc.perform(
+            MockMvcRequestBuilders.get(
+                "/api/management/v1/case/{caseDefinitionName}/{caseDefinitionVersion}/export",
+                caseDefinitionName, caseDefinitionVersion
+            )
+        ).andExpect(status().isOk)
+            .andExpect(MockMvcResultMatchers.jsonPath("$").isNotEmpty)
```
Comment (ivo-ritense) [inline case/src/test/kotlin/com/ritense/case/web/rest/CaseDefinitionResourceIntTest.kt:1182]: Does this work? Is the output json? Should we match on jsonpath?

File: document/src/main/kotlin/com/ritense/document/service/JsonSchemaDocumentDefinitionExportService.kt [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            )
+        }
+
+        return setOf(exportFile)
+    }
+
+    companion object {
+        const val PATH = "config/document/definition/%s.schema.json"
+        private val MAPPER = Mapper.INSTANCE.get()
+
```
Comment (ivo-ritense) [inline document/src/main/kotlin/com/ritense/document/service/JsonSchemaDocumentDefinitionExportService.kt:47]: Super nitpicking: i don't like the extra blank line.

File: document/src/test/kotlin/com/ritense/document/service/JsonSchemaDocumentDefinitionExportServiceIntTest.kt [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    private val resourceLoader: ResourceLoader,
+    private val documentDefinitionExportService: JsonSchemaDocumentDefinitionExportService
+) : BaseIntegrationTest() {
+
+    @Test
+    fun `should export document definition`(): Unit = runWithoutAuthorization {
+        val definition = documentDefinitionService.findLatestByName("person").orElseThrow()
+        val exportFiles = documentDefinitionExportService.export(definition.id())
+
+        val path = PATH.format(definition.id().name())
```
Comment (ivo-ritense) [inline document/src/test/kotlin/com/ritense/document/service/JsonSchemaDocumentDefinitionExportServiceIntTest.kt:41]: Would it be better to use a hardcoded path? you now sort of test the output by comparing it with itself.

File: document/src/test/kotlin/com/ritense/document/service/JsonSchemaDocumentDefinitionExportServiceIntTest.kt [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        val personDefinitionExport = exportFiles.singleOrNull {
+            it.path == path
+        }
+        assertThat(personDefinitionExport).isNotNull
+        requireNotNull(personDefinitionExport)
```
Comment (ivo-ritense) [inline document/src/test/kotlin/com/ritense/document/service/JsonSchemaDocumentDefinitionExportServiceIntTest.kt:46]: Seems like a duplicate of the line above.

File: case/src/main/kotlin/com/ritense/case/service/CaseDefinitionExportService.kt (hunk 1 above)
Comment (marijnritense) [reply case/src/main/kotlin/com/ritense/case/service/CaseDefinitionExportService.kt:32]: No, use automatically closes the stream at the end.

File: case/src/test/kotlin/com/ritense/case/web/rest/CaseDefinitionResourceIntTest.kt (hunk 3 above)
Comment (marijnritense) [reply case/src/test/kotlin/com/ritense/case/web/rest/CaseDefinitionResourceIntTest.kt:1182]: I guess all we're interested in at this stage is if the result is not empty.

**Pull request closed without merging**